esmvaltool/diag_scripts/climate_patterns/climate_patterns.py

```diff
@@ -0,0 +1,658 @@
+    land_frac: cube
+        gridded land fraction
+    area: str
+        area over which to calculate patterns
+
+    Returns
+    -------
+    slope_array : arr
+        array of grid cells with same shape as initial cube,
+        containing the regression slope
+    """
+    if area == "land":
+        # calculate average warming over land
+        tas_data = sf.area_avg_landsea(
+            tas, ocean_frac, land_frac, land=True, return_cube=False
+        )
+    else:
+        # calculate global average warming
+        tas_data = area_statistics(tas, 'mean').data
+
+    # Reshape cube for regression
+    cube_reshaped = cube_data.reshape(cube_data.shape[0], -1)
+
+    # Perform linear regression on valid values
+    model = sklearn.linear_model.LinearRegression(
+        fit_intercept=False, copy_X=True
+    )
+    model.fit(tas_data.reshape(-1, 1), cube_reshaped)
+
+    # Extract regression coefficients
+    slopes = model.coef_
+
+    # Reshape the regression coefficients back to the shape of the grid cells
+    slope_array = slopes.reshape(cube_data.shape[1:])
+
+    return slope_array
+
+
+def create_cube(tas_cube, ssp_cube, array, month_number, units=None):
+    """Create a new cube from existing metadata, and new aray data.
+
+    Parameters
+    ----------
+    tas_cube: cube
+        near-surface air temperature
+    ssp_cube: cube
+        cube of a given variable
+    array: array
+        output array from regression
+    month_number: int
+        month related to the regression array
+    units: str
+        units related to the regression variable
+
+    Returns
+    -------
+    cube: cube
+        cube filled with regression array and metadata
+
+    """
+    # assigning dim_coords
+    coord1 = tas_cube.coord(contains_dimension=1)
+    coord2 = tas_cube.coord(contains_dimension=2)
+    dim_coords_and_dims = [(coord1, 0), (coord2, 1)]
+
+    # assigning aux_coord
+    coord_month = iris.coords.AuxCoord(month_number, var_name="imogen_drive")
+    aux_coords_and_dims = [(coord_month, ())]
+
+    cube = sf.rename_variables(ssp_cube, has_orig_vars=False)
+
+    # creating cube
+    cube = iris.cube.Cube(
+        array,
+        units=units,
+        dim_coords_and_dims=dim_coords_and_dims,
+        aux_coords_and_dims=aux_coords_and_dims,
+        var_name=cube.var_name,
+        standard_name=cube.standard_name,
+    )
+
+    return cube
+
+
+def calculate_regressions(
+    anom_list,
+    area,
+    ocean_frac=None,
+    land_frac=None,
+    yrs=86
+):
+    """Facilitate the calculation of regression coeffs (climate patterns).
+
+    Also creates of a new cube of patterns per variable.
+
+    Parameters
+    ----------
+    anom_list : cubelist
+        cube list of variables as anomalies
+    ocean_frac: cube
+        gridded ocean fraction
+    land_frac: cube
+        gridded land fraction
+    area: str
+        area over which to calculate patterns
+    yrs : int
+        int to specify length of scenario
```

Comment:
Would it be possible to ensure the order of these parameters matches with the order they are defined in the function, please? 😊

Reply:
Sure! Resolved this in the latest commit